types/uint.go

```diff
@@ -107,6 +107,30 @@ func (u Uint) Mod(u2 Uint) Uint {
 	return Uint{mod(u.i, u2.i)}
 }
 
+// Incr increments the Uint by one
+func (u Uint) Incr() Uint {
+	return u.Add(OneUint())
+}
+
+// Decr decrements the Uint by one.
+// Decr will panic if the Uint is zero
+func (u Uint) Decr() Uint {
+	return u.Sub(OneUint())
+}
+
+// Cmp compares two Uints
+// If a > b, 1 is returned. If a < b, -1 is returned.
+// If a == b, 0 is returned.
+func (u Uint) Cmp(other Uint) int {
```

Comment:
To be protobuf compatible:

```suggestion
func (u Uint) Compare(other Uint) int {
```

Reply:
I think we should remove this altogether

Reply:
thats also fine 👍

Reply:
I'm removing this.